scalajslib/test/src/mill/scalajslib/HelloJSWorldTests.scala

```diff
@@ -124,7 +124,7 @@ object HelloJSWorldTests extends TestSuite {
       val module = HelloJSWorld.helloJsWorld(scalaVersion, scalaJSVersion)
       val jsFile =
         if (legacy) {
-          val task = if (optimize) module.fullOpt else module.fastOpt
+          val task = if (optimize) module.fullLinkJS else module.fastLinkJS
```

Comment:
This seems to be wrong. I think the `legacy: Boolean` switch is exactly there to test the `fullOpt` and `fastOpt` targets.